packages/app/src/cli/models/extensions/specifications/payments_app_extension_schemas/custom_credit_card_payments_app_extension_schema.ts

```diff
@@ -31,11 +36,61 @@ export const CustomCreditCardPaymentsAppExtensionSchema = BasePaymentsAppExtensi
       .optional(),
   })
 
+export interface CustomCreditCardPaymentsAppExtensionDeployConfigType extends BasePaymentsAppExtensionDeployConfigType {
+  ui_extension_registration_uuid?: string
+
+  // Following are overwritten as they are required for custom credit card extensions
+  start_refund_session_url: string
+  start_capture_session_url: string
+  start_void_session_url: string
+
+  // ConfirmationSchema
+  confirmation_callback_url?: string
+  supports_3ds: boolean
+
+  multiple_capture: boolean
+  checkout_hosted_fields?: string[]
+  ui_extension_handle?: string
+  encryption_certificate: string
+  checkout_payment_method_fields?: {
+    type: 'string' | 'number' | 'boolean'
+    required: boolean
+    key: string
+  }[]
+}
+
+export async function customCreditCardDeployConfigToCLIConfig(
+  config: CustomCreditCardPaymentsAppExtensionDeployConfigType,
+): Promise<
+  Omit<CustomCreditCardPaymentsAppExtensionConfigType, 'name' | 'type' | 'metafields' | 'targeting'> | undefined
+> {
+  return {
+    api_version: config.api_version,
+    ui_extension_registration_uuid: config.ui_extension_registration_uuid,
+    payment_session_url: config.start_payment_session_url,
+    refund_session_url: config.start_refund_session_url,
+    capture_session_url: config.start_capture_session_url,
+    void_session_url: config.start_void_session_url,
```

Comment:
same for these

Reply:
I assume `payment_session_url` is what the server expects right? can't we have that in the toml as well instead of `start_payment_session_url`?
(or viceversa)

Reply:
Just trying to understand if there is a reason to have different names for the same thing toml vs api and if we can unify everything to simplify the code.

Reply:
I think this naming was decided on with the tech design: https://docs.google.com/document/d/1Y2Ly5sw6MalZQP8w3yraQlOs8dK3UK38LVQvQz7fC_M/edit#heading=h.8coln95uazox

I agree with you, but we'll have to have these different config field names I'm afraid

Reply:
Let's discuss this in Slack